tests/test_cancellations.py

```diff
@@ -35,3 +38,39 @@ async def test_async_cancellation():
         with anyio.move_on_after(0.001):
             await pool.request("GET", "http://example.com")
         assert not pool.connections
+
+
+@pytest.mark.anyio
+async def test_h11_response_closed():
+    origin = httpcore.Origin(b"https", b"example.com", 443)
+    stream = httpcore.AsyncMockStream([])
+    async with httpcore.AsyncHTTP11Connection(origin, stream) as conn:
+        # mock
+        conn._state = HTTP11ConnectionState.ACTIVE
+
+        with anyio.CancelScope() as cancel_scope:
+            async with conn._state_lock:
+                cancel_scope.cancel()
+                await conn._response_closed()
+        assert conn._state != HTTP11ConnectionState.ACTIVE
+        assert conn._state in (HTTP11ConnectionState.IDLE, HTTP11ConnectionState.CLOSED)
```

Comment:
Okay so, challenge...

Tests ought to be against our public API. In this case, using `conn.request(...)` with a cancellation.